server/session_auth.go

```diff
@@ -242,23 +242,23 @@ func (a *authenticationService) StartServer(logger *zap.Logger) {
 }
 
 func (a *authenticationService) handleAuth(w http.ResponseWriter, r *http.Request,
-	retrieveUserID func(authReq *AuthenticateRequest) ([]byte, string, string, int)) {
+	retrieveUserID func(authReq *AuthenticateRequest) ([]byte, string, string, Error_Code, int)) {
```

Comment:
Do we need to return the error code number (400, 401, 500) etc given that we have an `Error_Code` that we can `switch` on to produce the correct HTTP error number?